snapshots/overlay/overlay.go

```diff
@@ -572,3 +605,23 @@ func supportsIndex() bool {
 	}
 	return false
 }
+
+// getMetacopyState returns metacopy state according to kernel support and
+// overlayfs mount options.
+func getMetacopyState(metacopy bool, userxattr bool) MetacopyState {
+	if _, err := os.Stat("/sys/module/overlay/parameters/metacopy"); err != nil {
+		return metacopyUnsupported
+	}
+
+	if !metacopy {
+		logrus.Infof("Metacopy disabled.")
+		return metacopyDisabled
```

Comment:
Our recommendation has been to use the system parameters here. If you had previously enabled then upgraded with this change, would it become disabled by default with no configuration change?

Reply:
Yes, with no configuration change metacopy will be disabled, even it was enabled by `CONFIG_OVERLAY_FS_METACOPY=Y` in Linux kernel. The `CONFIG_OVERLAY_FS_METACOPY` option is usually disabled by default, but yes, we can have customized installation or customized overlay.ko that contains `CONFIG_OVERLAY_FS_METACOPY = Y`.

Possibly we can do like this:
In case of metacopy=true in /etc/containerd/config.toml we add metacopy=on option to overlayfs mount point if metacopy is supported. Otherwise, don't add any option to overlayfs mount point and use system defaults. In this case it will be safe to use default configuration if `CONFIG_OVERLAY_FS_METACOPY=Y`

With such approach we will be unable to switch metacopy support off for overlayfs snapshotter if metacopy is enabled system-wide. But looks like this is not our goal.

I'll update the logic If such approach is ok.

Reply:
@rata Good idea, it would be bad to introduce string field in config or something like this to enable 3-state configuration. Will test and update PR. Thanks!

Reply:
Hi, I've updated the implementation. Now, if there is no metacopy option in /etc/containerd/config.toml the overlayfs module configuration is used, and containerd doesn't override system default. 
Also, in case of *bool config option for metacopy support it is better to use booleans in snapshotter config instead of enum, due to it simplify options conversion. 

@rata, @AkihiroSuda, @dmcgowan, could you have a look please.

Thank you in advance!